hyperon_das_atomdb/database.py

```diff
@@ -259,28 +252,32 @@ def _build_link(
         composite_type = [link_type_hash]
         composite_type_hash = [link_type_hash]
 
+        target_handles = []
         for target in targets:
             if not isinstance(target, dict):
-                raise ValueError('The target must be a dictionary')
-            if 'targets' not in target:
+                raise ValueError("The target must be a dictionary")
+            if "targets" not in target:
                 atom = self.add_node(target)
                 if atom is None:
                     return None
-                atom_hash = atom['composite_type_hash']
+                atom_hash = atom["composite_type_hash"]
                 composite_type.append(atom_hash)
             else:
                 atom = self.add_link(target, toplevel=False)
                 if atom is None:
                     return None
-                atom_hash = atom['composite_type_hash']
-                composite_type.append(atom['composite_type'])
+                atom_hash = atom["composite_type_hash"]
+                composite_type.append(atom["composite_type"])
             composite_type_hash.append(atom_hash)
-            targets_hash.append(atom['_id'])
+            target_handles.append(atom["_id"])
+            targets_hash.append(atom["_id"])
 
         handle = ExpressionHasher.expression_hash(link_type_hash, targets_hash)
 
         link: LinkT = {
             FieldNames.ID_HASH: handle,
+            "handle": handle,
+            "targets": target_handles,
```

Comment:
Shouldn't these be added to the `FieldNames` enum?

Reply:
No. FieldNames is supposed to held fields used in MongoDB. `handle` and `targets` aren't fields in MongoDB's documents. Yes, this whole fileld names and reserved names for document fields should be revised but we already have a card for this. https://github.com/singnet/das-query-engine/issues/75 and https://github.com/singnet/das-atom-db/issues/57